services/src/test/scala/cromwell/services/auth/impl/GithubAuthVendingActorSpec.scala

```diff
@@ -8,94 +8,81 @@ import cromwell.services.auth.GithubAuthVending.{
+import org.scalatest.prop.TableDrivenPropertyChecks
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class GithubAuthVendingActorSpec extends TestKitSuite with AnyFlatSpecLike with Matchers with Eventually {
+class GithubAuthVendingActorSpec
+    extends TestKitSuite
+    with AnyFlatSpecLike
+    with Matchers
+    with Eventually
+    with TableDrivenPropertyChecks {
 
+  final private val validUserToken = "valid_user_token"
   final private val githubAuthEnabledServiceConfig =
     ConfigFactory.parseString(s"""
                                  |enabled = true
                                  |auth.azure = true
                                  |ecm.base-url = "https://mock-ecm-url.org"
         """.stripMargin)
 
-  behavior of "GithubAuthVendingActor"
-
-  it should "return NoGithubAuthResponse if GithubAuthVending is disabled" in {
-    val serviceConfig = ConfigFactory.parseString(s"""
-                                                     |enabled = false
-                                                     |auth.azure = false
-      """.stripMargin)
-
-    val serviceRegistryActor = TestProbe()
-    val actor = system.actorOf(
-      Props(new GithubAuthVendingActor(serviceConfig, ConfigFactory.parseString(""), serviceRegistryActor.ref))
-    )
-
-    eventually {
-      serviceRegistryActor.send(actor, GithubAuthRequest("valid_user_token"))
-      serviceRegistryActor.expectMsg(NoGithubAuthResponse)
-    }
-  }
-
-  it should "return invalid configuration error if no ECM base url is found" in {
-    val serviceConfig = ConfigFactory.parseString(s"""
-                                                     |enabled = true
-                                                     |auth.azure = true
-      """.stripMargin)
-
-    val serviceRegistryActor = TestProbe()
-    val actor = system.actorOf(
-      Props(new GithubAuthVendingActor(serviceConfig, ConfigFactory.parseString(""), serviceRegistryActor.ref))
-    )
-
-    eventually {
-      serviceRegistryActor.send(actor, GithubAuthRequest("valid_user_token"))
-      serviceRegistryActor.expectMsg(
-        GithubAuthVendingFailure("Invalid configuration for service 'GithubAuthVending': missing 'ecm.base-url' value.")
-      )
-    }
-  }
+  private def githubAuthConfigWithoutEcmUrl(flag: Boolean): Config =
+    ConfigFactory.parseString(s"""
+                                 |enabled = $flag
+                                 |auth.azure = $flag
+        """.stripMargin)
 
-  it should "return Github token if found" in {
-    val serviceRegistryActor = TestProbe()
-    val actor = system.actorOf(
-      Props(
-        new TestGithubAuthVendingActor(githubAuthEnabledServiceConfig,
-                                       ConfigFactory.parseString(""),
-                                       serviceRegistryActor.ref
-        )
-      )
+  private val testCases = Table(
+    ("test name", "service config", "use TestEcmService class", "terra token", "response"),
```

Comment:
Can we add a case for "valid config / no token for user in ECM"?

Reply:
Do you think this is covered by the last test case `return failure message if ECM service returns non-successful response`? It seems that ECM would return a non-successful response if there was no Github token associated with the user and the last test case does simulate `getGithubAccessToken()` returning a Failed future which is what would happen in "no token for user in ECM" case too.